### PR TITLE
feat(dataproxy): expose error info to users

### PR DESCRIPTION
**Pull request by endor**
Data proxy already sends useful information with error responses. This PR now exposes that information to users.

### CODE REVIEW

File: packages/engine-core/src/data-proxy/errors/utils/responseToError.ts [hunk 1]
```diff
@@ -40,6 +53,24 @@ export async function responseToError(
...
   if (response.status >= 500) {
+    let body
+    try {
+      body = await response.json()
+    } catch (err) {
+      throw new ServerError(info)
+    }
+
+    if (typeof body?.EngineNotStarted?.reason === 'string') {
+      throw new ServerError(info, body?.EngineNotStarted?.reason)
```
Comment (Jolg42) [inline packages/engine-core/src/data-proxy/errors/utils/responseToError.ts:64]:
```suggestion
      throw new ServerError(info, body.EngineNotStarted.reason)
```
Nit: should work with less `?` here

File: packages/engine-core/src/data-proxy/errors/utils/responseToError.ts [hunk 2]
```diff
@@ -40,6 +53,24 @@ export async function responseToError(
...
+    } else if (typeof body?.EngineNotStarted?.reason === 'object') {
+      const keys = Object.keys(body?.EngineNotStarted?.reason)
+      if (keys.length > 0) {
+        const reason = body?.EngineNotStarted?.reason
```
Comment (Jolg42) [inline packages/engine-core/src/data-proxy/errors/utils/responseToError.ts:68]:
```suggestion
      const keys = Object.keys(body.EngineNotStarted.reason)
      if (keys.length > 0) {
        const reason = body.EngineNotStarted.reason
```
```suggestion
      throw new ServerError(info, body.EngineNotStarted.reason)
```
Nit: should work with less `?` here

**Review by Jolg42 — APPROVED** [review]
That looks useful indeed! Nice ✨

**Comment by janpio** [comment]
Is there a way to test this functionality?

**Pull request merged**